core/src/main/java/com/github/mizool/core/validation/CheckEnumValue.java

```diff
@@ -16,10 +16,12 @@
  */
 package com.github.mizool.core.validation;
 
+import java.util.List;
+
 import javax.validation.ConstraintValidator;
 import javax.validation.ConstraintValidatorContext;
 
-public class CheckEnumValue implements ConstraintValidator<EnumValue, String>
+public class CheckEnumValue implements ConstraintValidator<EnumValue, Object>
```

Comment:
before we adapt this check (and potentially others), we should try using `@EnumValue(...) List<String>` (BeanVal < 2.0) and `List<@EnumValue(...) String>` (as of BeanVal 2.0, using a JDK 8 language change allowing annotations on type usages and not only declarations).

Reply:
see also https://beanvalidation.org/proposals/BVAL-508/